src/components/Icon.tsx

```diff
@@ -0,0 +1,81 @@
+import { ComponentProps, ElementType, ReactNode } from "react";
+
+export type Color =
+  | "white"
+  | "black"
+  | "light"
+  | "dark"
+  | "primary"
+  | "link"
+  | "info"
+  | "success"
+  | "warning"
+  | "danger"
+  | "black-bis"
+  | "black-ter"
+  | "grey-darker"
+  | "grey-dark"
+  | "grey"
+  | "grey-light"
+  | "grey-lighter"
+  | "white-ter"
+  | "white-bis"
+  | "current"
+  | "inherit";
+
+export interface IconProps extends ComponentProps<"span"> {
+  /** Font Awesome icon class name */
+  icon: string;
+  textColor?: Color;
+  /**
+   * The Font Awesome icon should also have a size modifier.
+   * `https://docs.fontawesome.com/web/style/size`
+   */
+  size?: "small" | "medium" | "large";
+}
+
+export const Icon = ({ icon, textColor, size }: IconProps) => {
+  const baseClass = "icon";
+  const colorClass = textColor ? `has-text-${textColor}` : "";
+  const sizeClass = size ? `is-${size}` : "";
+
+  const classes = [baseClass, colorClass, sizeClass].filter(Boolean).join(" ");
+
+  return (
+    <span className={classes}>
+      <i className={icon} />
+    </span>
+  );
+};
+
+interface IconTextProps<T extends ElementType = "span"> {
+  /** The element type to render. Defaults to "span". */
```

Comment:
Not a huge fan of these comments. Seems obvious and inconsistent with the rest of the code